syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.env";
option java_outer_classname = "ReplayBufferProto";
option objc_class_prefix = "HLW";

package actor;

// The greeting service definition.
service Actor {
  rpc ServiceReady (ServiceReadyReq) returns (ServiceReadyResp) {}
  rpc Start (StartReq) returns (StartResp) {}
  rpc ReplicateModel (ReplicateModelReq) returns (ReplicateModelResp) {}
  rpc Terminate (TerminateReq) returns (TerminateResp) {}
}

message ServiceReadyReq {
  string requestId = 1;
}

message ServiceReadyResp {
  bool ready = 1;
  string requestId = 2;
}

message StartReq {
  string requestId = 1;
}

message StartResp {
  bool done = 1;
  string err = 2;
  string requestId = 3;
}

message Meta {
  string version = 3;
  string requestId = 4;
}

message ReplicateModelReq {
  oneof request {
    Meta meta = 1;
    bytes chunkData = 2;
  }
}

message ReplicateModelResp {
  bool done = 1;
  string err = 2;
  string requestId = 3;
}

message TerminateReq {
  string requestId = 1;
}

message TerminateResp {
  bool done = 1;
  string err = 2;
  string requestId = 3;
}
